app/assets/javascripts/course/components/CoursePage.js

```diff
@@ -108,9 +107,11 @@ class CoursePage extends React.Component {
     const subsection = this.state.displayedSubsection
     const component = this.state.displayedComponent
 
-    if (!component.is_complete) {
-      this.markComponentAsComplete(component)
-    } else if (!isLast(subsection.components, component)) {
+    if (subsection.is_complete) {
+      this.markSubsectionComplete(subsection)
+    }
```

Comment:
Wait why was this logic changed? Also, why are we marking a subsection complete when it's already complete?